packages/snjs/lib/services/api/mfa_service.ts

```diff
@@ -0,0 +1,169 @@
+import { Challenge, ChallengeKeyboardType, ChallengePrompt, ChallengeReason, ChallengeValidation } from '@Lib/challenges';
+import { ChallengeService, SNAlertService } from '..';
+import { PureService } from '../pure_service';
+import { SNApiService } from './api_service';
+import { PromptTitles, SessionStrings, SignInStrings } from './messages';
+import { HttpResponse, StatusCode } from './responses';
+
+export enum MfaStatus {
+  EnteredCode = 'mfa-entered-code',
+  NotRequired = 'mfa-not-required',
+  Ignored = 'mfa-ignored',
+
+  ErrorMissingCode = 'mfa-error-missing-code',
+  ErrorResponse = 'mfa-error-response',
+}
+
+export type MfaResultOld = {
+  status: MfaStatus.EnteredCode,
+  mfaKey: string,
+  enteredMfaCode: string,
+} | {
+  status: Exclude<MfaStatus, MfaStatus.EnteredCode>,
+  response: HttpResponse<unknown>,
+}
+
+export type MfaResult = {
+  status: MfaStatus.EnteredCode,
+  totpUuid: string,
```

Comment:
Wouldn't this be uuid of the UserSetting and not the totp?

Reply:
Good catch, will fix.